pontoon/base/tests/managers/test_translationmemory.py

```diff
@@ -0,0 +1,162 @@
+    )
+
+
+@pytest.fixture
+def tm_entry_short():
+    yield TranslationMemoryFactory.create(
+        source=u'a' * 50,
+    )
+
+
+@pytest.mark.django_db
+@patch('pontoon.base.models.TranslationMemoryEntryManager.postgres_levenshtein_ratio')
+@patch('pontoon.base.models.TranslationMemoryEntryManager.python_levenshtein_ratio')
+def test_levenshtein_ratio_string_below_255_chars(python_mock, postgresql_mock, tm_entry_short):
+    TranslationMemoryEntry.objects.minimum_levenshtein_ratio(
+        tm_entry_short.locale,
+        tm_entry_short.source,
+    )
+
+    assert not python_mock.called
+    assert postgresql_mock.call_args == call(
+        tm_entry_short.locale,
+        tm_entry_short.source,
+        0.7,
+        35,
+        71,
+    )
+
+
+@pytest.mark.django_db
+@patch('pontoon.base.models.TranslationMemoryEntryManager.postgres_levenshtein_ratio')
+@patch('pontoon.base.models.TranslationMemoryEntryManager.python_levenshtein_ratio')
+def test_levenshtein_ratio_string_equals_255_chars(python_mock, postgresql_mock, tm_entry_medium):
+    TranslationMemoryEntry.objects.minimum_levenshtein_ratio(
+        tm_entry_medium.locale,
+        tm_entry_medium.source,
+    )
+
+    assert not postgresql_mock.called
+    assert python_mock.call_args == call(
+        tm_entry_medium.locale,
+        tm_entry_medium.source,
+        0.7,
+        179,
+        364,
+    )
+
+
+@pytest.mark.django_db
+@patch('pontoon.base.models.TranslationMemoryEntryManager.postgres_levenshtein_ratio')
+@patch('pontoon.base.models.TranslationMemoryEntryManager.python_levenshtein_ratio')
+def test_levenshtein_ratio_string_above_255_chars(python_mock, postgresql_mock, tm_entry_long):
+    TranslationMemoryEntry.objects.minimum_levenshtein_ratio(
+        tm_entry_long.locale,
+        tm_entry_long.source,
+    )
+
+    assert not postgresql_mock.called
+    assert python_mock.call_args == call(
+        tm_entry_long.locale,
+        tm_entry_long.source,
+        0.7,
+        350,
+        714,
+    )
+
+
+@pytest.mark.django_db
+def test_levenshtein_ratio_both_implementations(tm_entry_short):
+    """
+    Check if postgresql and postgresql return the same results for strings lower than 255.
+    """
+    postgresql_queryset = list(
+        TranslationMemoryEntry.objects.postgres_levenshtein_ratio(
+            tm_entry_short.locale,
+            tm_entry_short.source,
+            0.7,
+            35,
+            71,
+        ).values_list('pk', 'source', 'target', 'quality')
+    )
+
+    python_queryset = list(
+        TranslationMemoryEntry.objects.python_levenshtein_ratio(
+            tm_entry_short.locale,
+            tm_entry_short.source,
+            0.7,
+            35,
+            71,
+        ).values_list('pk', 'source', 'target', 'quality')
+    )
+
+    expected_results = [
+        (tm_entry_short.pk, tm_entry_short.source, tm_entry_short.target, 100),
+    ]
+    assert postgresql_queryset == python_queryset == expected_results
```

Comment:
That's going to output confusing messages in case of a failure. I think it would be better to break it into 2 lines.